sed/loader/mpes/loader.py

```diff
@@ -254,18 +241,39 @@ def hdf5_to_array(
 
     # Delayed array for loading an HDF5 file of reasonable size (e.g. < 1GB)
 
+    # determine group length from per_electron column:
+    nelectrons = 0
+    for channel in channels:
+        if channel["format"] == "per_electron":
+            nelectrons = len(h5file[channel["group_name"]])
+            break
+    if nelectrons == 0:
+        raise ValueError("No 'per_electron' columns defined, or no hits found in file.")
```

Comment:
I added this, because I need to no the number of events in the file, to generate the per_file columns. I think empty files will also break the time_stamp functionality anyways, so they are currently not supported.